BritneyS/dnd-graphql
# Create query

## What does this PR do :question:
This PR includes the implementation of a successful GraphQL query of the D&D open api endpoint: http://dnd5eapi.co/api/classes/


## How to test it :microscope:
1. Check out this branch, and navigate to this project's root directory in the terminal
2. Install dependencies by running `npm install`
3. In the project's root directory, add a file called `.env`, and include the line `ENV_PORT=8000` in the file
4. Run `npm start`
5. In the terminal, nodemon should execute and the string `Apollo Server on http://localhost:8000/graphql` should be printed in the console (see screenshot)
6. In the browser, navigate to `http://localhost:8000/graphql`, and note that the GraphQL Playground environment should be running (see screenshot)
7. In the query window of the playground environment, query the following: 
```
{
  result {
   count
   results {
     name
    }
  }
}
```
Run the query, and note the expected response:

```
{
  "data": {
    "result": {
      "count": 12,
      "results": [
        {
          "name": "Barbarian"
        },
        {
          "name": "Bard"
        },
        {
          "name": "Cleric"
        },
        {
          "name": "Druid"
        },
        {
          "name": "Fighter"
        },
        {
          "name": "Monk"
        },
        {
          "name": "Paladin"
        },
        {
          "name": "Ranger"
        },
        {
          "name": "Rogue"
        },
        {
          "name": "Sorcerer"
        },
        {
          "name": "Warlock"
        },
        {
          "name": "Wizard"
        }
      ]
    }
  }
}
```
(see screenshot)


## Any background info you would like to include :white_check_mark:
N/A


## Screenshots (if applicable) :camera:
<img width="474" alt="Screen Shot 2019-05-11 at 2 24 51 PM" src="https://user-images.githubusercontent.com/8409475/57573645-b9ee5a00-73f8-11e9-918d-11b8c039bd3b.png">
<img width="1175" alt="Screen Shot 2019-05-11 at 2 25 27 PM" src="https://user-images.githubusercontent.com/8409475/57573647-beb30e00-73f8-11e9-901f-2edc87b208e4.png">
